### PR TITLE
fix issue 19523 - "undefined identifier" causes DMD crash

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by Basile-z** [comment]
I was not able to run the test locally on stable although the same patch worked on master so let's see...

**Comment by wilzbach** [comment]
As @Basile-z was a few seconds faster than me in finding the problem with excel-d, here's the link for others: https://github.com/kaleidicassociates/excel-d/issues/63

**Review by PetarKirov — APPROVED** [review]

**Review by jacob-carlborg — APPROVED** [review]

File: test/fail_compilation/b19523.d [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/b19523.d(12): Error: undefined identifier `SomeStruct`
+fail_compilation/b19523.d(13): Error: function `b19523.foo(int delegate() arg)` is not callable using argument types `(_error_)`
```
Comment (wilzbach) [inline test/fail_compilation/b19523.d]: Should `_error_` really appear in the user-facing output?

Comment (jacob-carlborg) [reply test/fail_compilation/b19523.d]: I would say ideally not.

**Pull request merged**